echopype/convert/parse_azfp.py

```diff
@@ -127,25 +90,36 @@ def load_AZFP_xml(self):
         """Parses the AZFP  XML file.
         """
 
-        def get_value_by_tag_name(tag_name, element=0):
-            """Returns the value in an XML tag given the tag name and the number of occurrences."""
-            return px.getElementsByTagName(tag_name)[element].childNodes[0].data
-
         xmlmap = fsspec.get_mapper(self.xml_path, **self.storage_options)
-        px = xml.dom.minidom.parse(xmlmap.fs.open(xmlmap.root))
-
-        # Retrieve integer parameters from the xml file
-        for old_name, new_name in XML_INT_PARAMS.items():
-            self.parameters[new_name] = int(get_value_by_tag_name(old_name))
-        # Retrieve floating point parameters from the xml file
-        for param in XML_FLOAT_PARAMS:
-            self.parameters[param] = float(get_value_by_tag_name(param))
-        # Retrieve frequency dependent parameters from the xml file
-        for old_name, new_name in XML_FREQ_PARAMS.items():
-            self.parameters[new_name] = [
-                float(get_value_by_tag_name(old_name, ch))
-                for ch in range(self.parameters["num_freq"])
-            ]
+        root = ET.parse(xmlmap.fs.open(xmlmap.root)).getroot()
+
+        for child in root.iter():
+            if len(child.attrib) > 0:
+                for key, val in child.attrib.items():
+                    self.parameters[child.tag + key].append(val)
```

Comment:
What do you think about changing `child.tag + key` to `child.tag + key.capitalize()` so that we have the same camelcase form that is then put through the same converting camelcase to lowercase+"_" procedure, so that at the output end in `self.parameters` we have identical strings? For example, right now the first key would be `InstrumentTypestring`, and if we do the above, we will get `instrument_type_string` -- this latter one would go very well with `instrument_type` for cross referencing what is what.

Reply:
Sounds great!, but one query - `sensors_flag` has `PressureSensorInstalled` as one attribute, should that be converted to `sensors_flag_pressure_sensor_installed` or `sensors_flag_pressuresensorinstalled`? What do you think about this? I think the latter one should be correct.

Reply:
hmmm, this is a tricky question. I think `sensors_flag_pressure_sensor_installed` is easier to read. What if we go with that one?

Reply:
Sure, we can go with `sensors_flag_pressure_sensor_installed`. This naming choice does make it more clear and readable.